src/main/scala/com.snowplowanalytics.snowplow.micro/Routing.scala

```diff
@@ -99,7 +72,62 @@ final class Routing(igluResolver: Resolver[IO], validationCache: ValidationCache
   }
 }
 
+final class InMemoryRouting(protected val igluResolver: Resolver[IO], protected val storage: InMemoryStorage)
+                           (implicit protected val lookup: RegistryLookup[IO]) extends BaseRouting[InMemoryStorage] {
+  private val inMemoryRoutes: HttpRoutes[IO] = HttpRoutes.of[IO] {
+    case GET -> Root / "micro" / "events" =>
+      Ok(storage.getGoodAndIncomplete.map(_.event.toJson(lossy = true)))
+    case (POST | GET) -> Root / "micro" / "all" =>
+      Ok(storage.getSummary())
+    case GET -> Root / "micro" / "good" =>
+      Ok(storage.filterGood(FiltersGood(None, None, None, None)))
+    case request @ POST -> Root / "micro" / "good" =>
+      request.as[FiltersGood].flatMap { filters =>
+        Ok(storage.filterGood(filters).asJson)
+      }
+    case GET -> Root / "micro" / "bad" =>
+      Ok(storage.filterBad(FiltersBad(None, None, None)))
+    case request @ POST -> Root / "micro" / "bad" =>
+      request.as[FiltersBad].flatMap { filters =>
+        Ok(storage.filterBad(filters))
+      }
+    case _ -> "micro" /: _ =>
+      NotFound("Supported endpoints: /micro/events, /micro/all, /micro/good, /micro/bad, /micro/reset, /micro/iglu, /micro/ui")
+  }
+
+  val routes: HttpRoutes[IO] = commonRoutes <+> inMemoryRoutes
+}
+
+final class SqliteRouting(protected val igluResolver: Resolver[IO],
+                          protected val storage: SqliteStorage)
+                         (implicit protected val lookup: RegistryLookup[IO]) extends BaseRouting[SqliteStorage] {
+  private val sqliteRoutes: HttpRoutes[IO] = HttpRoutes.of[IO] {
+    case GET -> Root / "micro" / "events" =>
+      storage.getEvents.flatMap(events => Ok(events))
+    case _ -> "micro" /: _ =>
+      NotFound("Supported endpoints: /micro/events, /micro/reset, /micro/iglu, /micro/ui")
+  }
+
+  val routes: HttpRoutes[IO] = commonRoutes <+> sqliteRoutes
+}
+
+object NoRoutes extends Http4sDsl[IO] {
+  val routes: HttpRoutes[IO] = HttpRoutes.of[IO] {
+    case _ -> "micro" /: _ => NotFound()
+  }
+}
+
 object Routing {
+  def create(config: Configuration.MicroConfig, storage: EventStorage, lookup: RegistryLookup[IO]): HttpRoutes[IO] = {
+    storage match {
+      case NoStorage => NoRoutes.routes
```

Comment:
The irony is not lost on me.

<img width="400" height="225" alt="image" src="https://github.com/user-attachments/assets/56f35fff-d0e4-46aa-95de-dd5b31c4baee" />

Reply:
:joy: